tools/trap/trap.c

```diff
@@ -288,6 +289,67 @@ static int extfsdump_recv(int dev_fd)
         return 0;
 }
 
+int dump_file_from_fs(int dev_fd, char *filename)
+{
+	int ret;
+	int fd;
+	int i, last_subdir;
+	char buf;
+	FILE *fp;
+	u_int64_t file_size;
+	char char_filesize[8];
+	char local_filename[CONFIG_SMARTFS_MAXNAMLEN + 9];
+
+	printf("Attempting to Dump File %s from Target Device\n", filename);
+
+	ret = write(dev_fd, filename, CONFIG_SMARTFS_MAXNAMLEN);
+	if (ret != CONFIG_SMARTFS_MAXNAMLEN) {
+		printf("Unable to send Dump File name to Target Device\n");
+		return -1;
+	}
+
+	file_size = 0;
+	for (int i = 0; i < 8; i++) {
+		ret = read(dev_fd, &char_filesize[i], sizeof(char_filesize[i]));
+		if (ret != sizeof(char_filesize[i])) {
+			printf("Unable to get file size from Target Device\n");
+			return -1;
+		}
+	}
+
+	file_size = (((u_int64_t)char_filesize[0] << 56) & 0xFF00000000000000U) | (((u_int64_t)char_filesize[1] << 48) & 0x00FF000000000000U) | (((u_int64_t)char_filesize[2] << 40) & 0x0000FF0000000000U) | (((u_int64_t)char_filesize[3] << 32) & 0x000000FF00000000U) | ((char_filesize[4] << 24) & 0x00000000FF000000U) | ((char_filesize[5] << 16) & 0x0000000000FF0000U) | ((char_filesize[6] <<  8) & 0x000000000000FF00U) | (char_filesize[7] & 0x00000000000000FFU);
+
+	printf("\nReceiving file %s of size %lu from Target Device\n\n", filename, file_size);
+
+	i = 0;
+	last_subdir = -1;
+	while (filename[i] != '\0') {
+		/* We dont want the file to be saved in the root folder, we want it in the current folder */
+		if (filename[i] == '/') {
+			last_subdir = i;
+		}
+		i++;
+	}
+
+	filename = filename + last_subdir + 1;
+	snprintf(local_filename, CONFIG_SMARTFS_MAXNAMLEN + 9, "%s%s", "DumpFile_", filename);
+
+	fp = fopen(local_filename, "w");
+	if (fp == NULL) {
+		printf("Unable to open file %s to save logs chain dump\n", filename);
+		return -1;
+	}
+
+	ret = get_dump(dev_fd, fp, file_size);
```

Comment:
```
get_dump(FILE *fp)
{
    fopen();
    ...do_something
    fclose();
}
```
get_dump seems that get data from somewhere, so it should handle fp alone

Reply:
It is better that handle open & close in one function, it is always better option. you may know it enough, why don't you modify it?

Reply:
Yes, it can be done.
I will check and create a different commit for that if confirmed